pkg/api/validation/validation.go

```diff
@@ -2477,6 +2477,19 @@ func validateServiceFields(service *api.Service) field.ErrorList {
 			allErrs = append(allErrs, field.Invalid(fieldPath, val, "must be a list of IP ranges. For example, 10.240.0.0/24,10.250.0.0/24 "))
 		}
 	}
+
+	// Validate ExternalTraffic and HealthCheckNodePort have legal value.
+	if service.Spec.HealthCheckNodePort < 0 {
+		allErrs = append(allErrs, field.Invalid(specPath.Child("healthCheckNodePort"), service.Spec.HealthCheckNodePort,
+			"can not set HealthCheckNodePort to negative value"))
+	}
+	if service.Spec.ExternalTraffic != "" &&
```

Comment:
Updates...[Per discussion above](https://github.com/kubernetes/kubernetes/pull/41162#discussion_r102340291), "" might be allowed for service that does not have LoadBalancer or NodePort type.